### PR TITLE
Migrate `logic.js` to TypeScript

### LINKED ISSUE #308: Migrate logic.js to TypeScript
## Summary
This issue tracks the migration of logic.js to TypeScript.

## Scope
Migrate `src/logic/logic.js` to TypeScript.

## Urgency
Required to finish the TypeScript migration.

### PR DESCRIPTION
**Pull request by jdtzmn**
Closes #308.

### CODE REVIEW

File: src/logic/logic.ts [hunk 1]
```diff
@@ -3,14 +3,34 @@
...
+
+let opcodes: {
+  [key: number]: OpStructure;
+};
 
 const maxCost = 20000;
 const maxLength = 1000;
 
-function parseUvarint(array) {
+export function parseUvarint(array: Uint8Array) {
```
Comment (jasonpaulos) [inline src/logic/logic.ts:33]: Could you add a named return type of `[numberFound: number, size: number]` to this function?

File: src/logic/logic.ts [hunk 2]
```diff
@@ -88,7 +114,7 @@ function readByteConstBlock(program, pc) {
   return [size, byteArrays];
 }
 
-function readPushIntOp(program, pc) {
+function readPushIntOp(program: Uint8Array, pc: number) {
```
Comment (jasonpaulos) [inline src/logic/logic.ts:117]: Can you add a named return type here too?

**Review by jasonpaulos — APPROVED** [review]
I left two comments about nice-to-have type annotations, but other than that everything here looks good! Feel free to merge after those additions.

File: src/logic/logic.ts (hunk 1 above)
Comment (jdtzmn) [reply src/logic/logic.ts:33]: Good idea!

**Pull request merged**